docs/proposals/27-project-group.md

```diff
@@ -0,0 +1,197 @@
+- Enhance projects so that shoot control planes can be restricted to certain geographical regions ([`SeedBinding`s](#seedbinding))
+- Share those, so called, seed bindings to different projects that logically belong together in a concept called [`ProjectGroup`s](#projectgroup)
+- Introduce new role for project group administrators, that are allowed to add projects to a project group
+
+### Non-Goals
+
+- Let users exclusively bind seeds on their own
+
+## Proposal
+
+#### ProjectGroup
+
+A ProjectGroup API will be introduced that abstracts certain information over one or more projects. With this, it will become possible to homogeneously share resources between projects.
+
+Such a ProjectGroup manifest looks like this:
+```yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: ProjectGroup
+metadata:
+  name: group1
+spec:
+  namespace: gardenprojectgroup-group1
+  owner:
+    apiGroup: rbac.authorization.k8s.io
+    kind: User
+    name: john.doe@example.com
+  members:
+  - apiGroup: rbac.authorization.k8s.io
+    kind: User
+    name: alice.doe@example.com
+    role: admin
+  projects:
+    - Project1
+    - Project2
+```
+
+The main idea behind `ProjectGroup`s is that they, like projects, have their own namespace.
+Every resource that is shareable (at first only `SeedBinding`s) in this namespace, will be shared to all projects in this project group.
+
+This sharing mechanism will be achieved by the `ProjectGroup` controller.
+It will copy the objects from the project group namespace to the namespaces of the projects that belong to the project group.
+
+When the object in the project group namespace is changed, the controller will immediately update the objects in the namespaces of the projects.
+
+To let a user in a project know that such an object is being kept in sync with an object in a project group, a `projectgroup.gardener.cloud/copied-from: <projectgroup-name>` label will be added to it.
+
+This copying feature can become useful for future use cases, for example for sharing infrastructure credentials across different projects, by copying `SecretBinding`s.
+
+It will not be possible to add non-existent project to a project group. This also means that if a project is deleted that is part of a project group, the controller should remove it.
+
+The `.spec.namespace` field in `ProjectGroup` will behave equally as the `Project` `.spec.namespace` field.
+
+##### Permissions
+
+Anybody should be able to create a project group.
+
+But adding a project to a project group is a sensitive operation.
+
+If anyone could add any project to their project group, a bad actor could add a project of a third party to their project group and restrict them from using any seed except the ones bound in the bad actors project group.
+
+To circumvent a scenario like this, a new role, called `project-group-assigner` will be introduced that allows people with this role to add the project to a project group, in which they have the same role.
+
+#### SeedBinding
```

Comment:
What should happen to existing shoot clusters when a `SeedBinding` is created in a project? For example, what happens to shoot clusters not conforming to the `SeedBinding`.

What should happen if a `SeedBinding` changes invalidating the existing shoot scheduling decisions?

Reply:
> What should happen to existing shoot clusters when a `SeedBinding` is created in a project? For example, what happens to shoot clusters not conforming to the `SeedBinding`.

Regarding your comment on control plane migration, I think this should be the solution to the problems that you are mentioning here.

> What should happen if a `SeedBinding` changes invalidating the existing shoot scheduling decisions?

What do you mean by invalidating existing shoot scheduling decisions?

Reply:
> Regarding your comment on control plane migration, I think this should be the solution to the problems that you are mentioning here.

As of now, control plane migration implies a downtime of the control plane. (Duration depends on infrastructure and some boundary conditions, e.g. `etcd` size.)

I ask because I am hesitant to use control plane migration as an automatic solution here. Furthermore, I would like to see the boundary conditions mentioned explicitly. In other words:
- If we think that creating a seed binding will automatically trigger a control plane migration it should be noted in the document.
- If updates to seed binding should not be possible if the invalidate a shoot scheduling decision it needs to be mentioned.
- If we only apply the seed binding during scheduling/control plane migration, but ignore it at all other times we should at least note that it is possible that the decision where to schedule a control plane to might change over time due to updates to the seed binding and someone else needs to take care of cleaning that up.

> What do you mean by invalidating existing shoot scheduling decisions?

If I have seeds `A` and `B`, create a seed binding in my project matching seed `A` and create a shoot it will be scheduled to seed `A`. However, if I now realise my seed binding is wrong and change it to match `B` instead the previous scheduling decision concerning my shoot cluster is invalid. What should happen in that case?